equinix/helpers_device.go

```diff
@@ -95,6 +97,48 @@ func getNetworkInfo(ips []*packngo.IPAddressAssignment) NetworkInfo {
 	return ni
 }
 
+func getMetalGoNetworkType(device *metalv1.Device) (*string, error) {
+
+	pgDevice := packngo.Device{}
+	res, err := device.MarshalJSON()
+	if err != nil {
+		json.Unmarshal(res, pgDevice)
+		networkType := pgDevice.GetNetworkType()
+		return &networkType, nil
+	}
+	return nil, err
+}
+
+func getMetalGoNetworkInfo(ips []metalv1.IPAssignment) NetworkInfo {
+	ni := NetworkInfo{Networks: make([]map[string]interface{}, 0, 1)}
+	for _, ip := range ips {
+		network := map[string]interface{}{
+			"address": *ip.Address,
+			"gateway": *ip.Gateway,
+			"family":  *ip.AddressFamily,
+			"cidr":    *ip.Cidr,
+			"public":  *ip.Public,
+		}
+		ni.Networks = append(ni.Networks, network)
+
+		// Initial device IPs are fixed and marked as "Management"
+		if !ip.HasManagement() || *ip.Management {
```

Comment:
I'm not 100% sure what the goal of this logic is: `management == null || management == true` ?

We have additional nil-safe helpers generated that may be able to simplify these conditions (as with the HasPublic check): `ip.GetManagement()` https://pkg.go.dev/github.com/equinix-labs/metal-go@v0.19.1/metal/v1#IPReservation.GetManagement

If I am misinterpreting the logic, GetManagementOk() may be more fitting.

`*ip.AddressFamily == int32(4)` could be `ip.GetAddressFamily() == 4`

Reply:
Same concern:

`!ip.HasPublic() || *ip.Public` sounds like `ip.public == nil || ip.public == true`, I think the intention is (pseudo-code) `ip.public != nil && ip.public == true`.

`if ip.GetPublic() {}` should handle that case.

Reply:
Yeah no idea what the aim was with these conditions but the `Get` helpers should be analogous to what the packngo version of this function was doing.